pkg/clusteragent/autoscaling/controller.go

```diff
@@ -30,11 +30,12 @@ type Controller struct {
 	context   context.Context
 
 	// Fields available to child controllers
-	ID        string
-	Client    dynamic.Interface
-	Lister    cache.GenericLister
-	Workqueue workqueue.RateLimitingInterface
-	IsLeader  func() bool
+	ID             string
+	Client         dynamic.Interface
+	Lister         cache.GenericLister
+	Workqueue      workqueue.RateLimitingInterface
+	AutoscalerHeap *HashHeap
```

Comment:
That's not something generic to any controller, should not be part of the generic `Controller`. Generally speaking, it's an internal part of the controller and there's no probably no reason to take is as creation parameter.